operator/cilium_node.go

```diff
@@ -223,6 +226,7 @@ func startSynchronizingCiliumNodes(ctx context.Context, nodeManager allocator.No
 		// then there isn't any event handler set for CiliumNodes events.
 		if withKVStore && kvStoreSyncHandler != nil {
 			<-connectedToKVStore
+			log.Info("Connected to the KVStore, syncing CiliumNodes to the KVStore")
 			for processNextWorkItem(kvStoreQueue, kvStoreSyncHandler) {
 			}
```

Comment:
What about this one?

Reply:
I find that confusing. Could we add a comment to explain why? I'm not sure I understand why line 216 needs a goroutine and line 230 doesn't.